test/new-e2e/pkg/test/pulumi_provisioner.go

```diff
@@ -0,0 +1,97 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package test
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"reflect"
+
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/runner"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/infra"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+)
+
+const (
+	PulumiProvisionerDefaultID = "pulumi"
+)
+
+type PulumiEnvRunFunc[Env any] func(ctx *pulumi.Context, env *Env) error
+
+type PulumiProvisioner[Env any] struct {
+	id        string
+	runFunc   PulumiEnvRunFunc[Env]
+	configMap runner.ConfigMap
+}
+
+var (
+	_ TypedProvisioner[any] = &PulumiProvisioner[any]{}
+	_ Provisioner           = &PulumiProvisioner[any]{}
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❓ question**
Should this be 
```suggestion
	_ UntypedProvisioner           = &PulumiProvisioner[any]{}
```
?